client/src/pages/CreateWorkout/components/AddExercise.jsx

```diff
@@ -0,0 +1,158 @@
+import React from "react";
+import ExerciseCard from "@client/src/components/ExerciseCard/ExerciseCard";
+import { useState } from "react";
+
+const AddExercise = () => {
+  const [exercises, setExercises] = useState([]);
+  const [exercise, setExercise] = useState({
+    name: "",
+    // duration: { minutes: "", seconds: "" },
+    minutes: "",
+    seconds: "",
+    weight: "",
+    reps: "",
+    sets: "",
+  });
+
+  const handleChange = (e) => {
+    // if (e.target.name === "minutes" || e.target.name === "seconds") {
+    //   setExercise({
+    //     ...exercise,
+    //     duration: { ...exercise.duration, [e.target.name]: e.target.value },
+    //   });
+    // } else {
+    //   setExercise({ ...exercise, [e.target.name]: e.target.value });
+    // }
+    setExercise({ ...exercise, [e.target.name]: e.target.value });
```

Comment:
awesome job generalizing this handlechange.

One note is that when you are setting state based on the previous state, you should use a version of setState that takes in a function like so:

`setExercise(prevExercise -> ({ ...prevExercise, [e.target.name]: e.target.value }))`

This way, if 2 things change simultaneously or really close to each other - you are able to catch both changes. You can read more about why this is better & safer here: https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state